### ISSUE TITLE
random() with weights>>1 does not re-select at each end of track

### ISSUE THREAD

**Issue by martinkirch** [savonet/liquidsoap#1468]
It seems that `random()`'s `weights` sets the number of tracks played from the selected source before the next random pick.

**To Reproduce**
````
src = random([
   single("your favorite 1-minute track.flac"),
   single("your favorite 1-minute track.flac")
], weights=[6,4])

output.prefered(mksafe(src))
````
In this script's log,  `[random_8220:3] Switch to ...` happens at best every 4 or 6 tracks (or any multiple of 4 or 6).

I discovered the bug using `[60, 40]` : it switched every few hours.

**Expected behavior**
According to the documentation, I expected it to randomly pick a source after each track (from any source), with a weighted probability.

**Version details**
 - OS: Linux (Debian and Ubuntu server)
 - Version 1.4.3 installed via opam

**Comment by toots** [comment]
Thanks for this report. I'll look at it asap for the `1.4.4` release.

**Comment by toots** [comment]
Fixed with 8b156278c5c0c7331794344d3631c8d0c29dc704 !
